src/axom/sidre/core/Group.cpp

```diff
@@ -2367,12 +2367,20 @@ Group* Group::walkPath(std::string& path, bool create_groups_in_path)
       }
       else if(create_groups_in_path)
       {
-        group_ptr = group_ptr->createGroup(*iter);
-
-        if(group_ptr == nullptr)
+        if(m_is_list)
```

Comment:
Design question: Should the modification for lists happen here, inside of `walkPath`? or should the caller be aware of list paths, and possibly send in an empty string for `path`?